### ISSUE TITLE
Why Window.print() is removed?

### ISSUE THREAD

**Issue by Delagen** [Microsoft/TypeScript#22917]
**TypeScript Version:**  2.8.1

**Search Terms:** 

**Code**

```ts
window.print()
```

**Expected behavior:** No errors.

**Actual behavior:** TypeScript error missing property print in Window

**Playground Link:** https://www.typescriptlang.org/play/index.html#src=window.print()
**Related Issues:**

**Comment by Delagen** [comment]
67c7b368e6dd2b5980ea9eafa9b0944459e4eb09

**Comment by forestmaxime** [comment]
Issue is present in 2.8.0 also. The commit also removes functions that exists in standard. It would be a good idea to redo a full code review on this one.

**Comment by mhegazy** [comment]
PRs welcomed. You can find more information about contributing lib.d.ts fixes at https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md#contributing-libdts-fixes.

**Comment by Delagen** [comment]
git under windows produces unpredictable amount of changes, sorry for so many commits. I need to strip changelog to single line

**Comment by Delagen** [comment]
Thanks. Sorry I don,t read carefully instriction, that describe separate process for 2 files

**Comment by evil-shrike** [comment]
Also methods `onbeforeprint` and `onafterprint` were moved from `Window` into `WindowEventHandlers` which isn't related to `Window`. 
So it's a breaking change in 2.8.
Code
```
	window.onbeforeprint = beforePrint;
	window.onafterprint = afterPrint;
```
start failing to compile:
(533,11): error TS2339: Property 'onbeforeprint' does not exist on type 'Window'.  
(534,11): error TS2339: Property 'onafterprint' does not exist on type 'Window'.

**Comment by mhegazy** [comment]
@evil-shrike please file a new ticket for that one and will get it fixed.

**Comment by mhegazy** [comment]
> Issue is still present in TypeScript 2.8.1

yes. the fix is scheduled to be released in 2.8.2

**Comment by Kurtz1993** [comment]
Thanks @mhegazy !

**Comment by mangeshdesai26** [comment]
Thanks @mhegazy

**Comment by ps04639** [comment]
`(window as any).print();`

use above line if not using typescript 2.8.2.

**Comment by Delagen** [comment]
@ps04639 You are wrong If you think that author of issue does not read TS docs

**Comment by simonprice33** [comment]
(window as any).print(); looks to get around the issue but is still not great as when it creates the js file it resolves to window.print() 

I am working on the latest available TypeScript via Nuget and the issue is still present today

**Comment by Delagen** [comment]
@simonprice33 What wrong in you answer, and what you expect?

**Comment by ExDeaDguY** [comment]
@simonprice33 I believe at the time of posting this, the latest TS version is 2.8.3. If you are using VSCode (currently ships/updates with 2.8.1) - you can update the TS version to the latest with the command "npm install -g typescript@2.8.3". Follow the voted answer here for more information: [Stack Overflow - TS Version](https://stackoverflow.com/questions/39668731/what-typescript-version-is-visual-studio-code-using-how-to-update-it)

**Comment by JaSpr** [comment]
I'm using ts 2.9.1 and still getting this error.

**Comment by ChamilaMaddumage** [comment]
I am using this code snippet to print my PowerBI embedded report using Angular7 and getting following error.


  let report = powerbi.get(reportContainer);

  //Trigger the print dialog for your browser.
  report .print()
    .then(function (result) {
      console.log(result);
    })
    .catch(function (errors) {
      console.log(errors);
    });

Error : [ts] property 'print' does not exist on type 'Embed'

**Comment by simonprice33** [comment]
@ChamilaMaddumage try `(report as any).print()` and see how you get on
